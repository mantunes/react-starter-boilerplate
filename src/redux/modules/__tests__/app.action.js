import configureMockStore from 'redux-mock-store';

import * as actions from '../app';

const mockStore = configureMockStore([]);
const store = mockStore({});

beforeEach(() => {
  store.clearActions();
});

it('ADD_NAME (Rui Miguel)', () => {
  store.dispatch(actions.addName('Rui Miguel'));
  expect(store.getActions()).toMatchSnapshot();
});

it('ADD_NAME (empty)', () => {
  store.dispatch(actions.addName(''));
  expect(store.getActions()).toMatchSnapshot();
});

it('SUBTRACT_NAME (0)', () => {
  store.dispatch(actions.removeName(0));
  expect(store.getActions()).toMatchSnapshot();
});

it('SUBTRACT_NAME (empty)', () => {
  store.dispatch(actions.removeName());
  expect(store.getActions()).toMatchSnapshot();
});

// NÃ£o consigo reproduzir o resultado do fetch()
//global.fetch = jest.fn().mockImplementation(
//  () => Promise.resolve(
//    'json': ({
//      userId: 1,
//      id: 1,
//      title: 'sunt aut facere repellat provident occaecati',
//      body:
//        'quia et suscipit suscipit recusandae consequuntur rem eveniet architecto',
//    }),
//  ),
//);
//
//it('GET_POST', async () => {
//  await store.dispatch(actions.getPost());
//  expect(store.getActions()).toMatchSnapshot();
//});
