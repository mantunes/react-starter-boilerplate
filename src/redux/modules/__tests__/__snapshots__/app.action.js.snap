// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADD_NAME (Rui Miguel) 1`] = `
Array [
  Object {
    "payload": Object {
      "name": "Rui Miguel",
    },
    "type": "ADD_NAME",
  },
]
`;

exports[`ADD_NAME (empty) 1`] = `
Array [
  Object {
    "payload": Object {
      "name": "",
    },
    "type": "ADD_NAME",
  },
]
`;

exports[`SUBTRACT_NAME (0) 1`] = `
Array [
  Object {
    "payload": Object {
      "index": 0,
    },
    "type": "SUBTRACT_NAME",
  },
]
`;

exports[`SUBTRACT_NAME (empty) 1`] = `
Array [
  Object {
    "payload": Object {
      "index": undefined,
    },
    "type": "SUBTRACT_NAME",
  },
]
`;
